### PR TITLE
Code cleanup NonNull related changes

### PR DESCRIPTION
**Pull request by zhljen**

### CODE REVIEW

File: metacat-common-server/src/main/java/com/netflix/metacat/common/server/events/MetacatUpdateTablePreEvent.java [hunk 1]
```diff
@@ -51,8 +52,8 @@ public MetacatUpdateTablePreEvent(
         @Nonnull @NonNull final QualifiedName name,
         @Nonnull @NonNull final MetacatRequestContext requestContext,
         @Nonnull @NonNull final Object source,
-        @Nonnull @NonNull final TableDto oldTable,
-        @Nonnull @NonNull final TableDto currentTable
+        @Nullable final TableDto oldTable,
```
Comment (tgianos) [inline metacat-common-server/src/main/java/com/netflix/metacat/common/server/events/MetacatUpdateTablePreEvent.java]: Now that these are nullable you basically need to go to every place these events are used and check for null too...

**Comment by zhljen** [comment]
After discussion, I reverted this change and marked as TODO.

**Comment by zhljen** [comment]
Revert back the !=null removals.

File: metacat-common/src/main/java/com/netflix/metacat/common/dto/BaseDto.java [hunk 2]
```diff
@@ -35,7 +35,7 @@
  */
 //TODO: All DTO's should be READ-ONLY
 public abstract class BaseDto implements Serializable {
-    private static MetacatJsonLocator metacatJsonLocator = new MetacatJsonLocator();
+    protected static final MetacatJsonLocator METACAT_JSON_LOCATOR = new MetacatJsonLocator();
```
Comment (tgianos) [inline metacat-common/src/main/java/com/netflix/metacat/common/dto/BaseDto.java]: any reason this is static rather than a member?

Comment (zhljen) [inline metacat-common/src/main/java/com/netflix/metacat/common/dto/BaseDto.java]: Don't want to create instances in the deserializeObjectNode/ serializeObjectNode BaseDTO.

**Pull request merged**